lib/AppInfo/Application.php

```diff
@@ -51,29 +52,30 @@ public function register(IRegistrationContext $context): void {
 
 		// Wire the interface to our decorator and implementation
 		$context->registerService(ISentryReporter::class, static function (ContainerInterface $c) {
-			/** @var Config $config */
-			$config = $c->get(Config::class);
 			/** @var SentryReporterAdapter $inner */
 			$inner = $c->get(SentryReporterAdapter::class);
+			return new RecursionAwareReporter($inner);
+		});
+		$context->registerCrashReporter(ISentryReporter::class);
+		$context->registerMiddleware(PerformanceMonitoringMiddleware::class, true);
+		$context->registerEventListener(AddContentSecurityPolicyEvent::class, CustomCspListener::class);
+		$context->registerInitialStateProvider(DsnProvider::class);
+	}
+
+	public function boot(IBootContext $context): void {
+		Util::addScript('sentry', 'sentry');
 
+		$context->injectFn(static function(Config $config) {
 			// Now it's time to connect Sentry
 			$dsn = $config->getDsn();
 			if ($dsn !== null) {
 				initSentry([
 					'dsn' => $dsn,
 					'release' => $config->getServerVersion(),
+					'traces_sample_rate' => 1.0,
```

Comment:
Best would be to have this configurable :)

Reply:
Configurable and/or derived from the log level (debug->1.0, fatal->0.1)